pkgs/by-name/qq/qqmusic/package.nix

```diff
@@ -0,0 +1,140 @@
+{
+  fetchurl,
+  stdenv,
+  autoPatchelfHook,
+  makeWrapper,
+  lib,
+  makeDesktopItem,
+  copyDesktopItems,
+  # QQ Music dependencies
+  alsa-lib,
+  at-spi2-atk,
+  at-spi2-core,
+  cairo,
+  cups,
+  dbus,
+  expat,
+  gdk-pixbuf,
+  glib,
+  gtk3,
+  libdbusmenu,
+  libglvnd,
+  libpulseaudio,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  pciutils,
+  udev,
+  xorg,
+  ...
+}:
+################################################################################
+# Mostly based on qqmusic-bin package from AUR:
+# https://aur.archlinux.org/packages/qqmusic-bin
+################################################################################
+let
+  libraries = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    cairo
+    cups
+    dbus
+    expat
+    gdk-pixbuf
+    glib
+    gtk3
+    libdbusmenu
+    libglvnd
+    libpulseaudio
+    mesa
+    nspr
+    nss
+    pango
+    pciutils
+    udev
+    xorg.libX11
+    xorg.libxcb
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXScrnSaver
+    xorg.libXtst
+  ];
+in
+stdenv.mkDerivation {
+  pname = "qqmusic";
+  version = "1.1.5";
+  src = fetchurl {
+    url = "https://dldir1.qq.com/music/clntupate/linux/deb/qqmusic_1.1.5_amd64_.deb";
+    sha256 = "sha256-wTtO80S8o4/EeVCvBzaN4xtI3+mShTjNHpq41exfP+g=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    copyDesktopItems
+  ];
+  buildInputs = libraries;
+
+  unpackPhase = ''
+    ar x $src
+    tar xf data.tar.xz
+  '';
+
+  postInstall = ''
+    mkdir -p $out
+    cp -r opt/qqmusic $out/opt
+    cp -r usr/* $out/
+
+    rm -rf $out/opt/swiftshader
+    ln -sf ${libglvnd}/lib $out/opt/swiftshader
+
+    mkdir -p $out/bin
+    makeWrapper $out/opt/qqmusic $out/bin/qqmusic \
+      --argv0 "qqmusic" \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath libraries}"
+
+    # Hex patch
+    # 1. Fix orphaned processes
+    # 2. Fix search
+    local _subst="
```

Comment:
This looks a bit magical. Is there an explanation for what this does?

Reply:
This is directly copied from the [PKGBUILD of AUR qqmusic-bin package](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=qqmusic-bin). I personally have no idea what it does, other than the existing comments.

Reply:
Sounds like we need to ask the Arch maintainers what that's for, then

Reply:
I removed the hex patch for now. The app itself still works fine.